src/DynamoCore/Interfaces/IWorkspaceModel.cs

```diff
@@ -7,6 +7,7 @@ namespace Dynamo.Interfaces
     public interface IWorkspaceModel
     {
         string Name { get; }
+        string FileName { get; set; }
```

Comment:
I'm not sure, that we can make so.
Because in `CustomNodePathRemapper` FileName set directly. I.e.

```c#
ICustomNodeWorkspaceModel def;
var res = customNodeManager.TryGetFunctionWorkspace(id, this.isTestMode, out def);
var newPath = Path.Combine(newDirectoryPath, Path.GetFileName(def.FileName));
def.FileName = newPath;
```